src/components/Navigation/index.jsx

```diff
@@ -16,18 +18,36 @@ class Navigation extends Component {
 
   onChange(event) {
     const { name, value } = event.target;
-    const { history, platform } = this.props;
+    const {
+      history, platform, benchmark, timeRange,
+    } = this.props;
     let newPlatform;
     let newBenchmark;
-    if (name === 'platform') {
-      newPlatform = value;
-      newBenchmark = 'overview';
-    } else {
-      newPlatform = platform;
-      newBenchmark = value;
+    let newTimeRange;
+
+    switch (name) {
+      case 'platform':
+        newPlatform = value;
+        newBenchmark = 'overview';
+        newTimeRange = timeRange;
+        break;
+      case 'benchmark':
+        newPlatform = platform;
+        newBenchmark = value;
+        newTimeRange = timeRange;
+        break;
+      case 'timeRange':
+        newPlatform = platform;
+        newBenchmark = benchmark;
+        newTimeRange = value;
+        break;
+      default:
+        newPlatform = 'win10';
+        newBenchmark = 'overview';
+        newTimeRange = CONFIG.default.timeRange;
     }
     // eslint-disable-next-line react/prop-types
-    history.push(`/${newPlatform}/${newBenchmark}`);
+    history.push(`/${newPlatform}/${newBenchmark}/${newTimeRange}`);
```

Comment:
The `DataControl` component will need to be able to push history, thus, we will need the `withRouter` that is used as part of the `export default`.